### DISCUSSION TITLE
How to remove first pre tag from nested pre tag.

### DISCUSSION THREAD

**Discussion by umschaudhary** [remarkjs/remark#947] [Q&A] [answered]
React markdown added two pre tag pre > pre which added a white background color. I want to remove it. 
<img width="1264" alt="Screen Shot 2022-01-25 at 17 16 39" src="https://user-images.githubusercontent.com/31402318/150969689-4c5cf18b-ce68-4ae9-ab5d-89711078c8f4.png">

**Comment by wooorm** [comment, marked as answer]
The don’t add it. `react-markdown` doesn’t add it. The way you configure it does.
Also: see https://github.com/remarkjs/.github/blob/main/support.md for how to ask proper questions

**Reply by Anthony-Gaudino** [reply]
Please see https://github.com/remarkjs/remark/discussions/684#discussioncomment-616455 and https://github.com/remarkjs/remark/discussions/684#discussioncomment-4080496 for solutions.

**Comment by soykje** [comment]
Hi :wave: I'm facing quite the same issue... @wooorm you were talking about _configuration_ could you please be more explicit? There is actually a `pre` tag wrapper apparently generated by `react-markdown` (as detailed [here](https://github.com/remarkjs/react-markdown/blob/bd8e53b4969a0a6f5cfd0fb1d4fe5d97d2cfa630/changelog.md#600---2021-04-15) if I'm not wrong?). I'm looking for a way to remove it, as it breaks my layout... :confused: 

I tried a simple 
```javascript
// within components prop
code: ({ children, className, inline }) => inline ? <code>{children}</code> : <p>{children}</p>
```
which gives me at the end something like
```html
<pre>
  <p>Lorem ipsum dolor sit amet...</p>
</pre>
```

Thx in advance for your help! :pray:

**Reply by wooorm** [reply]
That’s not what this thread is about. This is about nested pre tags. A.k.a., two.

For what you’re doing: I have no clue why you’d want to turn off (block) code. How do you mean “breaks my layout”? You can style things with CSS.
While I don’t understand what you want to do and why, well, similar to a `code` component that you have, there’s also a `pre` component that you could change. There are also several options on ignoring elements (such as `disallowedElements`).

**Reply by soykje** [reply]
Hello, and thx for your quick answer :+1: Sorry, I understood the thread's topic as being about the `pre` wrapper (the "first pre tag")... :confused:
To explain myself better (I hope!), I was actually using `prism-react-renderer` to display my code blocks, with an old version of `react-markdown` (v5). After upgrading I noticed a `pre` wrapper for `code` component (as explained above), which brings two problems for me:
1. I apparently cannot style it using my current stack (CSS-In-JS, Styled Components)
2. I get a weird DOM, with two nested `pre`, as my custom block code component already has one

So this is not about turning off the block code, but juste about how I can use a custom component in this particular case (with generated and apparently non optional wrapper). I'll check the `disallowedElements` prop to see if that helps, anyway. Thank you!

**Reply by wooorm** [reply]
https://github.com/remarkjs/react-markdown#use-custom-components-syntax-highlight

**Reply by WeiGrand** [reply]
I also encountered this issue and I found out a simple workaround for this by adding a `pre` handler like this:
```jsx
 <ReactMarkdown
  components={{
    // return the children instead of wrapping the children with `<pre></pre>`
    pre({children}) {
      return children
    },
    code({node, inline, className, children, ...props}) {
      const match = /language-(\w+)/.exec(className || '')
      return !inline && match ? (
        <div className='not-prose'>
          <CodeBlock language={match[1]} {...props}>{children}</CodeBlock>
        </div>
      ) : (
        <code {...props} className={className}>
          {children}
        </code>
      )
    }
  }}
></ReactMarkdown>
```

**Reply by aditbhartia** [reply]
@WeiGrand's solution worked exactly for what I needed. Thank you! Maybe this can be documented somewhere or clarified in documentation? Without the pre handler, my default styling breaks within Markdown components.
